test/TestProjects/MixAPIVersion-Cadl/Generated/MixApiVersionClient.cs

```diff
@@ -0,0 +1,338 @@
+        /// <include file="Docs/MixApiVersionClient.xml" path="doc/members/member[@name='GetPets(RequestContext)']/*" />
+        public virtual Pageable<BinaryData> GetPets(RequestContext context = null)
+        {
+            return GetPetsImplementation("MixApiVersionClient.GetPets", context);
+        }
+
+        private Pageable<BinaryData> GetPetsImplementation(string diagnosticsScopeName, RequestContext context)
+        {
+            return PageableHelpers.CreatePageable(CreateEnumerable, ClientDiagnostics, diagnosticsScopeName);
+            IEnumerable<Page<BinaryData>> CreateEnumerable(string nextLink, int? pageSizeHint)
+            {
+                do
+                {
+                    var message = string.IsNullOrEmpty(nextLink)
+                        ? CreateGetPetsRequest(context)
+                        : CreateGetPetsNextPageRequest(nextLink, context);
+                    var page = LowLevelPageableHelpers.ProcessMessage(_pipeline, message, context, "value", "nextLink");
+                    nextLink = page.ContinuationToken;
+                    yield return page;
+                } while (!string.IsNullOrEmpty(nextLink));
+            }
+        }
+
+        internal HttpMessage CreateDeleteRequest(string name, RequestContext context)
+        {
+            var message = _pipeline.CreateMessage(context, ResponseClassifier204);
+            var request = message.Request;
+            request.Method = RequestMethod.Delete;
+            var uri = new RawRequestUriBuilder();
+            uri.Reset(_endpoint);
+            uri.AppendPath("/pets/Pet/", false);
+            uri.AppendPath(name, true);
+            uri.AppendQuery("api-version", _apiVersion, true);
+            request.Uri = uri;
+            request.Headers.Add("Accept", "application/json");
+            return message;
+        }
+
+        internal HttpMessage CreateReadRequest(int petId, RequestContext context)
+        {
+            var message = _pipeline.CreateMessage(context, ResponseClassifier200304);
+            var request = message.Request;
+            request.Method = RequestMethod.Get;
+            var uri = new RawRequestUriBuilder();
+            uri.Reset(_endpoint);
+            uri.AppendPath("/pets/", false);
+            uri.AppendPath(petId, true);
+            uri.AppendQuery("api-version", _apiVersion, true);
+            request.Uri = uri;
+            request.Headers.Add("Accept", "application/json");
+            return message;
+        }
+
+        internal HttpMessage CreateCreateRequest(RequestContent content, RequestContext context)
+        {
+            var message = _pipeline.CreateMessage(context, ResponseClassifier200);
+            var request = message.Request;
+            request.Method = RequestMethod.Post;
+            var uri = new RawRequestUriBuilder();
+            uri.Reset(_endpoint);
+            uri.AppendPath("/pets", false);
+            uri.AppendQuery("apiVersion", "2022-07-09", true);
```

Comment:
But this `apiVersion` parameter defined in cadl won't be exposed to external?

Reply:
It is a special case, the operation is bind to a special apiversion, so it will define in the operation as @query apiVersion: "2022-07-09". So it is not the same as the one in clientOption, we must pass the exact value in Create***Request method.